### DISCUSSION TITLE
How to use haskell-template with docker?

### DISCUSSION THREAD

**Discussion by aleeusgr** [srid/haskell-template#65] [Ideas]
I am figuring this out, plz feel free to jump in.
My goal is to build a REST endpoint and dockerize it.
1. I started with stack new, and using this [template](https://github.com/SIDANWhatever/plutus-cborhex-automation/issues/3#issuecomment-1279909016)
2. Adding nix support: clone&test haskell-template✅, merge it into my program ❌
 

working [nix support](https://www.tweag.io/blog/2022-06-02-haskell-stack-nix-shell/)

**Comment by aleeusgr** [comment]
16:09 $ nix run
warning: Git tree '/home/alex/workshop/hello-spock' is dirty
error: builder for '/nix/store/nfp0q5601qnn2wyhxc4m1jha73lkcym9-hello-spock-0.1.0.0.drv' failed with exit code 1;
       last 10 log lines:
       >     It could refer to
       >        either ‘Prelude.atomicModifyIORef'’,
       >               imported from ‘Prelude’ at app/Main.hs:2:8-11
       >               (and originally defined in ‘Relude.Lifted.IORef’)
       >            or ‘Data.IORef.atomicModifyIORef'’,
       >               imported from ‘Data.IORef’ at app/Main.hs:9:1-17
       >               (and originally defined in ‘GHC.IORef’)
       >    |
       > 27 |               visitorNumber <- liftIO $ atomicModifyIORef' ref $ \i -> (i+1, i+1)
       >    |                                         ^^^^^^^^^^^^^^^^^^
       For full logs, run 'nix log /nix/store/nfp0q5601qnn2wyhxc4m1jha73lkcym9-hello-spock-0.1.0.0.drv'.

**Something wrong with imports, .cabal is huge - maybe I can remove some of deps?**

**Reply by srid** [reply]
You can remove relude to fix this

https://srid.ca/haskell-template/relude

**Reply by aleeusgr** [reply]
Awesome!
I got to `hello-world`
thank you
PS I got to docker setup..

**Comment by aleeusgr** [comment]
I got a suggestion that running nix manager in a docker container is a suboptimal solution I should avoid; building a docker container using nix-build is the recommended solution I should try first

**Reply by srid** [reply]
Checkout [`pkgs.dockerTools.buildImage`](https://nix.dev/tutorials/building-and-running-docker-images)

**Reply by aleeusgr** [reply]
that is exactly the tool I was advised to use. Thank you for confirming!
My next goal is to reproduce hello-world with haskell-template
